plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/listener/DeployVersionMetaDataListener.java

```diff
@@ -49,22 +52,45 @@ private boolean isAbstractProject(Item item){
 		return item instanceof AbstractProject;
 	}
 	
-	private boolean isExistDeployVersionParameter(ParametersDefinitionProperty paramDefProp){
-		if (null != paramDefProp){
-			for(ParameterDefinition pDef : paramDefProp.getParameterDefinitions()){
-				if (pDef.getName().equals(Constants.DEPLOY_META_DATA))
-					return true;
-			}
+	private boolean isExistDeployVersionParameter(ParametersDefinitionProperty paramsDefinitionProperty){
+		return (findDeployMetaDataParameter(paramsDefinitionProperty) != null);
+	}
+	
+	private boolean isEqualApplicationName(ParametersDefinitionProperty paramsDefinitionProperty, DeployVersionMetaDataPublisher dataPublisher){
+		DeployMetaDataParameterDefinition parameterDefinition = findDeployMetaDataParameter(paramsDefinitionProperty);
+		if (null != parameterDefinition){
+			return dataPublisher.getDeployedAppName().equals(parameterDefinition.getApplicationName());
 		}
 		return false;
 	}
 	
-	private void addPublisherProperty(AbstractProject prj, ParametersDefinitionProperty paramDefProp, String applicationName){
+	private DeployMetaDataParameterDefinition findDeployMetaDataParameter(ParametersDefinitionProperty paramsDefinitionProperty){
+		if (null != paramsDefinitionProperty){
+			for(ParameterDefinition parameterDefinition : paramsDefinitionProperty.getParameterDefinitions()){
+				if (parameterDefinition instanceof DeployMetaDataParameterDefinition)
+					return (DeployMetaDataParameterDefinition)parameterDefinition;
+			}
+		}
+		return null;	
```

Comment:
i think it is the simplest way to solve case 'Item not found'.
But in another way it is possible to throw exception.
How better?